core/trino-main/src/test/java/io/trino/sql/query/QueryAssertions.java

```diff
@@ -389,11 +389,19 @@ public QueryAssert returnsEmptyResult()
          */
         public QueryAssert isFullyPushedDown()
         {
-            checkState(!(runner instanceof LocalQueryRunner), "testIsFullyPushedDown() currently does not work with LocalQueryRunner");
-
             // Compare the results with pushdown disabled, so that explicit matches() call is not needed
             verifyResultsWithPushdownDisabled();
 
+            return isFullyPushedDownRegardlessOfResults();
+        }
+
+        /**
+         * Verifies query is fully pushed down
+         */
+        public QueryAssert isFullyPushedDownRegardlessOfResults()
```

Comment:
nit: maybe instead add a modifier method `skippingResultsCheck` (like `skippingTypesCheck` we already have)

Reply:
https://github.com/trinodb/trino/pull/8052